src/cudamatrix/cu-vector.cc

```diff
@@ -51,9 +52,11 @@ Real VecVec(const CuVectorBase<Real> &a,
   if (CuDevice::Instantiate().Enabled()) {    
     Timer tim;
 
-    result = cublas_dot(a.Dim(), a.Data(), 1, b.Data(), 1);
+    cublasStatus_t stat;
+    stat = cublas_dot(handle, a.Dim(), a.Data(), 1, b.Data(), 1, &result);
 
-    CU_SAFE_CALL(cublasGetError());
+    //CU_SAFE_CALL(cublasGetError());
+    CU_SAFE_CALL(stat);
```

Comment:
CU_SAFE_CALL is not really designed to be called on a variable.  Instead, wrap the function call itself in CU_SAFE_CALL:
CU_SAFE_CALL(cublas_dot(handle, a.Dim(), a.Data(), 1, b.Data(), 1, &result));
Also, please remove rather than comment out deleted lines.